src/opticalglass/glassfactory.py

```diff
@@ -42,6 +45,49 @@ def register_glass(
     if not isinstance(medium, OpticalMedium):
         raise TypeError('medium must be an instance of OpticalMedium')
     _custom_glass_registry[key] = medium
+    if medium.catalog_name() not in _cat_names:
+        _cat_names.append(medium.catalog_name())
+        _cat_names_uc.append(medium.catalog_name().upper())
+
+
+class CustomGlassCatalog:
+    def __init__(self, cat):
+        self.catalog_name = cat
+        self.glass_list = [
+            # should return (gname_decode, gname, catalog) but gname_decode
+            # may not be defined for custom glasses. 
+            # gname_decode is supposed to be group_num, prefix, suffix. 
+            (('__NA__', '', ''), name, cat) for name, cat in _custom_glass_registry.keys()
+            if cat == cat
+        ]
+
```

Comment:
`self.glass_list` is used in 
https://github.com/mjhoptics/ray-optics/blob/2f8b5159ed7a6a7111ef73bba8ee338524358e79/src/rayoptics/seq/medium.py#L232-L235

We may not want to search substituting glass for custom glass, but without changing `ray-optics`, I was only able to have this dirty fix.
Modification in `ray-optics` should lead a better implementation, but I hope this is OK for now.